src/client.ts

```diff
@@ -82,22 +88,59 @@ export class APIClient {
     }
   }
 
-  public async getGroupsForUser(id) {
+  /**
+   * Iterates each application resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationships
+   */
+  public async iterateApplications(
+    iteratee: ResourceIteratee<OktaApplication>,
+  ): Promise<void> {
+    const apps: OktaApplication[] = [];
+    await this.oktaClient.listApplications().each((e) => {
+      apps.push(e);
+    });
+
+    for (const app of apps) {
+      await iteratee(app);
+    }
```

Comment:
Does the `.each` function accept an async function as an argument, so we can do this?

```suggestion
    await this.oktaClient.listApplications().each(async (e) => {
      await iteratee(app);
    });
```

Reply:
Yes! So much better, thank you. I'll do that for all of them, of course.